python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1299 @@
+        # rotation angle. Get boresight rotation angle from exposure
+        # visitInfo
+        try:
+            boresightRotAngle_rad = exposure.visitInfo.boresightRotAngle.asRadians()
+        except (AttributeError, RuntimeError):
+            self.log.warning(
+                "Could not get boresight rotation angle from visitInfo. Setting OCS coordinates to NaN."
+            )
+            boresightRotAngle_rad = 0.0
+            use_ocs = False
+        else:
+            use_ocs = True
+
+        if use_ocs:
+            # Rotate CCS coordinates to OCS using boresight rotation angle
+            cos_rot = np.cos(boresightRotAngle_rad)
+            sin_rot = np.sin(boresightRotAngle_rad)
+            # Note: field angles in degrees, but rotation angle is in radians
+            # Convert field angles to radians for rotation
+            thx_ccs_rad = np.deg2rad(field_x_list)
+            thy_ccs_rad = np.deg2rad(field_y_list)
+            # Apply rotation: OCS = R(rot) * CCS
+            donutData["thx_ocs"] = np.rad2deg(thx_ccs_rad * cos_rot - thy_ccs_rad * sin_rot)
+            donutData["thy_ocs"] = np.rad2deg(thx_ccs_rad * sin_rot + thy_ccs_rad * cos_rot)
+        else:
+            donutData["thx_ocs"] = [np.nan] * numDonuts
+            donutData["thy_ocs"] = [np.nan] * numDonuts
+
+        # th_n and th_w: Telescope North/West coordinates
+        # These can be calculated from sky coordinates and observatory position
+        # For now, leave as NaN if we can't calculate them
+        donutData["th_n"] = [np.nan] * numDonuts
+        donutData["th_w"] = [np.nan] * numDonuts
+
+        # Create the table
+        donutTable = QTable(donutData)
+
+        # Add units matching standard format
+        donutTable["coord_ra"].unit = u.rad  # Use radians like standard
+        donutTable["coord_dec"].unit = u.rad  # Use radians like standard
+        # Note: centroidX and centroidY are kept as plain floats to avoid
+        # Point2D constructor issues
+        donutTable["fx"].unit = u.deg
+        donutTable["fy"].unit = u.deg
+
+        # Add comprehensive visitInfo metadata using utility function
+        try:
+            donutTable = addVisitInfoToCatTable(exposure, donutTable)
+        except (AttributeError, RuntimeError) as e:
+            # Missing visitInfo/attributes or runtime errors
+            # (e.g., from DonutSizeCorrelator)
+            self.log.warning(
+                "Could not extract comprehensive visitInfo metadata: %s. Using minimal visitInfo structure.",
+                e,
+            )
+            # Provide minimal visitInfo structure
+            try:
+                visitId = int(exposure.visitInfo.id) if exposure.visitInfo else self.UNKNOWN_INT
+            except (AttributeError, ValueError, TypeError):
+                self.log.warning("Could not extract visit ID from exposure.visitInfo, using fallback -1")
+                visitId = self.UNKNOWN_INT
```

Comment:
This should never fail, so I'd just rely on it working rather than nesting `try` blocks.